test/unit/specs/core/edgeNetwork/sendEdgeNetworkRequestFactory.spec.js

```diff
@@ -392,4 +392,36 @@ describe("sendEdgeNetworkRequestFactory", () => {
       sendEdgeNetworkRequest({ payload, action })
     ).toBeRejectedWithError("Invalid XDM");
   });
+
+  it("returns the object from lifecycle::onBeforeRequest", () => {
+    lifecycle.onBeforeRequest.and.callFake(({ onResponse }) => {
+      onResponse(() => [{ a: 1 }]);
+      return Promise.resolve();
+    });
+
+    return expectAsync(
+      sendEdgeNetworkRequest({ payload, action })
+    ).toBeResolvedTo({ a: 1 });
+  });
+
+  it("returns the object from lifecycle::onResponse", () => {
+    lifecycle.onResponse.and.returnValue([{ a: 1 }]);
```

Comment:
The return value should probably be a promise resolved with the array to simulate actual behavior.

Reply:
yes, thank you